custom_components/tuya_local/devices/tuya_000004rime_weather_station.yaml

```diff
@@ -0,0 +1,586 @@
+            value: On
+            mask: 000000FF00000000000000
+          - dps_val: 0
+            value: Off
+            mask: 000000FF00000000000000
+      - id: 108
+        optional: true
+        type: base64
+        name: 3-day Forecast Weather
+        persist: true
+        mapping:
+          - mask: 00000000FF000000000000
+          - dps_val: 1
+            value: On
+            mask: 00000000FF000000000000
+          - dps_val: 0
+            value: Off
+            mask: 00000000FF000000000000
+      - id: 108
+        optional: true
+        type: base64
+        name: Local Temperature&Humidity
+        persist: true
+        mapping:
+          - mask: 0000000000FF0000000000
+          - dps_val: 1
+            value: On
+            mask: 0000000000FF0000000000
+          - dps_val: 0
+            value: Off
+            mask: 0000000000FF0000000000
+      - id: 108
+        optional: true
+        type: base64
+        name: Date
+        persist: true
+        mapping:
+          - mask: 000000000000FF00000000
+          - dps_val: 1
+            value: On
+            mask: 000000000000FF00000000
+          - dps_val: 0
+            value: Off
+            mask: 000000000000FF00000000
+      - id: 108
+        optional: true
+        type: base64
+        name: Week
+        persist: true
+        mapping:
+          - mask: 00000000000000FF000000
+          - dps_val: 1
+            value: On
+            mask: 00000000000000FF000000
+          - dps_val: 0
+            value: Off
+            mask: 00000000000000FF000000
+      - id: 108
+        optional: true
+        type: base64
+        name: Hourly Forecast Weather
+        persist: true
+        mapping:
+          - mask: 0000000000000000FF0000
+          - dps_val: 1
+            value: On
+            mask: 0000000000000000FF0000
+          - dps_val: 0
+            value: Off
+            mask: 0000000000000000FF0000
+      - id: 108
+        optional: true
+        type: base64
+        name: Sensor 1 Temperature&Humidity
+        persist: true
+        mapping:
+          - mask: 000000000000000000FF00
+          - dps_val: 1
+            value: On
+            mask: 000000000000000000FF00
+          - dps_val: 0
+            value: Off
+            mask: 000000000000000000FF00
+      - id: 108
+        optional: true
+        type: base64
+        name: Sensor 2 Temperature&Humidity
+        persist: true
+        mapping:
+          - mask: 00000000000000000000FF
+          - dps_val: 1
+            value: On
+            mask: 00000000000000000000FF
+          - dps_val: 0
+            value: Off
+            mask: 00000000000000000000FF
+  - entity: sensor
+    category: diagnostic
+    name: Alarm Settings
+    dps:
+      - id: 109
+        optional: true
+        type: boolean
+        name: sensor
+      - id: 109
+        optional: true
+        type: base64
+        name: Payload
+  - entity: sensor
+    category: diagnostic
+    name: Errors
```

Comment:
For a Fault sensor, it should be a binary_sensor, with class "problem", and mapping zero to false, and all other values to true.  The raw value can be stored on a fault_code extra attribute on the same entity, or primary entity.